### PR TITLE
Update Android IAP in C# project to Godot 4.3

### PR DESCRIPTION
**Pull request by JavierStark**
Attempt to update Android IAP in C# project to version 4.3 which was marked as needing an update in [#697]

**Changes**:
- Dotnet target framework net472 -> net6.0
- Correct use of Variants with AsType()
- Signal connection using SignalName and callable creation with Callable.From()

### CODE REVIEW

**Review by Calinou — APPROVED** [review]
Thanks!

File: mono/android_iap/Main.cs [hunk 1]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
                 _label.Text += $"\n\n\nTest item SKU: {TestItemSku}";
 
                 // No params.
-                _payment.Connect(nameof(GooglePlayBilling.Connected), this, nameof(OnConnected));
+                _payment.Connect(GooglePlayBilling.SignalName.Connected,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:31]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.Connected,
```

File: mono/android_iap/Main.cs [hunk 2]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
+                // Response ID (int), Debug message (string).
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledgementError, 
+                    Callable.From<int,string>(OnPurchaseAcknowledgementError));
                 // Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseConsumed), this, nameof(OnPurchaseConsumed));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseConsumed, 
+                    Callable.From<string>(OnPurchaseConsumed));
                 // Response ID (int), Debug message (string), Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseConsumptionError), this, nameof(OnPurchaseConsumptionError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseConsumptionError,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:61]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.PurchaseConsumptionError,
```

File: mono/android_iap/Main.cs [hunk 3]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
-                // Response ID (int), Debug message (string), Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseAcknowledgementError), this, nameof(OnPurchaseAcknowledgementError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledged, 
+                    Callable.From<string>(OnPurchaseAcknowledged));
+                // Response ID (int), Debug message (string).
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledgementError, 
+                    Callable.From<int,string>(OnPurchaseAcknowledgementError));
                 // Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseConsumed), this, nameof(OnPurchaseConsumed));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseConsumed,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:58]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.PurchaseConsumed,
```

File: mono/android_iap/Main.cs [hunk 4]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
+                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryError, 
+                    Callable.From<int,string, string[]>(OnSkuDetailsQueryError));
                 // Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseAcknowledged), this, nameof(OnPurchaseAcknowledged));
-                // Response ID (int), Debug message (string), Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseAcknowledgementError), this, nameof(OnPurchaseAcknowledgementError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledged, 
+                    Callable.From<string>(OnPurchaseAcknowledged));
+                // Response ID (int), Debug message (string).
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledgementError,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:55]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledgementError,
```

File: mono/android_iap/Main.cs [hunk 5]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
+                    Callable.From<Array>(OnSkuDetailsQueryCompleted));
                 // Response ID (int), Debug message (string), Queried SKUs (string[]).
-                _payment.Connect(nameof(GooglePlayBilling.SkuDetailsQueryError), this, nameof(OnSkuDetailsQueryError));
+                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryError, 
+                    Callable.From<int,string, string[]>(OnSkuDetailsQueryError));
                 // Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseAcknowledged), this, nameof(OnPurchaseAcknowledged));
-                // Response ID (int), Debug message (string), Purchase token (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseAcknowledgementError), this, nameof(OnPurchaseAcknowledgementError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledged,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:52]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.PurchaseAcknowledged,
```

File: mono/android_iap/Main.cs [hunk 6]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseError), this, nameof(OnPurchaseError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseError, 
+                    Callable.From<int,string>(OnPurchaseError));
                 // SKUs (Dictionary[]).
-                _payment.Connect(nameof(GooglePlayBilling.SkuDetailsQueryCompleted), this, nameof(OnSkuDetailsQueryCompleted));
+                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryCompleted, 
+                    Callable.From<Array>(OnSkuDetailsQueryCompleted));
                 // Response ID (int), Debug message (string), Queried SKUs (string[]).
-                _payment.Connect(nameof(GooglePlayBilling.SkuDetailsQueryError), this, nameof(OnSkuDetailsQueryError));
+                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryError,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:49]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryError,
```

File: mono/android_iap/Main.cs [hunk 7]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
-                _payment.Connect(nameof(GooglePlayBilling.PurchasesUpdated), this, nameof(OnPurchasesUpdated));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchasesUpdated, 
+                    Callable.From<Array>(OnPurchasesUpdated));
                 // Response ID (int), Debug message (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseError), this, nameof(OnPurchaseError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseError, 
+                    Callable.From<int,string>(OnPurchaseError));
                 // SKUs (Dictionary[]).
-                _payment.Connect(nameof(GooglePlayBilling.SkuDetailsQueryCompleted), this, nameof(OnSkuDetailsQueryCompleted));
+                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryCompleted,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:46]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.SkuDetailsQueryCompleted,
```

File: mono/android_iap/Main.cs [hunk 8]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
-                _payment.Connect(nameof(GooglePlayBilling.ConnectError), this, nameof(OnConnectError));
+                _payment.Connect(GooglePlayBilling.SignalName.ConnectError, 
+                    Callable.From<int,string>(OnConnectError));
                 // Purchases (Dictionary[]).
-                _payment.Connect(nameof(GooglePlayBilling.PurchasesUpdated), this, nameof(OnPurchasesUpdated));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchasesUpdated, 
+                    Callable.From<Array>(OnPurchasesUpdated));
                 // Response ID (int), Debug message (string).
-                _payment.Connect(nameof(GooglePlayBilling.PurchaseError), this, nameof(OnPurchaseError));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchaseError,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:43]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.PurchaseError,
```

File: mono/android_iap/Main.cs [hunk 9]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
-                _payment.Connect(nameof(GooglePlayBilling.Disconnected), this, nameof(OnDisconnected));
+                _payment.Connect(GooglePlayBilling.SignalName.Disconnected, 
+                    Callable.From(OnDisconnected));
                 // Response ID (int), Debug message (string).
-                _payment.Connect(nameof(GooglePlayBilling.ConnectError), this, nameof(OnConnectError));
+                _payment.Connect(GooglePlayBilling.SignalName.ConnectError, 
+                    Callable.From<int,string>(OnConnectError));
                 // Purchases (Dictionary[]).
-                _payment.Connect(nameof(GooglePlayBilling.PurchasesUpdated), this, nameof(OnPurchasesUpdated));
+                _payment.Connect(GooglePlayBilling.SignalName.PurchasesUpdated,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:40]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.PurchasesUpdated,
```

File: mono/android_iap/Main.cs [hunk 10]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
...
-                _payment.Connect(nameof(GooglePlayBilling.Connected), this, nameof(OnConnected));
+                _payment.Connect(GooglePlayBilling.SignalName.Connected, 
+                    Callable.From(OnConnected));
                 // No params.
-                _payment.Connect(nameof(GooglePlayBilling.Disconnected), this, nameof(OnDisconnected));
+                _payment.Connect(GooglePlayBilling.SignalName.Disconnected, 
+                    Callable.From(OnDisconnected));
                 // Response ID (int), Debug message (string).
-                _payment.Connect(nameof(GooglePlayBilling.ConnectError), this, nameof(OnConnectError));
+                _payment.Connect(GooglePlayBilling.SignalName.ConnectError,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:37]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.ConnectError,
```

File: mono/android_iap/Main.cs [hunk 11]
```diff
@@ -27,27 +28,38 @@ public override void _Ready()
                 _label.Text += $"\n\n\nTest item SKU: {TestItemSku}";
 
                 // No params.
-                _payment.Connect(nameof(GooglePlayBilling.Connected), this, nameof(OnConnected));
+                _payment.Connect(GooglePlayBilling.SignalName.Connected, 
+                    Callable.From(OnConnected));
                 // No params.
-                _payment.Connect(nameof(GooglePlayBilling.Disconnected), this, nameof(OnDisconnected));
+                _payment.Connect(GooglePlayBilling.SignalName.Disconnected,
```
Comment (Calinou) [inline mono/android_iap/Main.cs:34]:
```suggestion
                _payment.Connect(GooglePlayBilling.SignalName.Disconnected,
```

**Comment by JavierStark** [comment]
I found some problems when building for Android and after uploading into Google Play Store. I will fix it shortly

**Pull request merged**